### PR TITLE
Update SimCLR tutorial to use low-level blocks

### LINKED ISSUE #553: Rewrite SimCLR Tutorial with Low-Level Building Blocks
Rewrite the SimCLR tutorial with low-level building blocks as we want to remove mid-level blocks (#493)
Specifically:

- [x] Rewrite SimCLR Model
- [x] Remove dependency on `lightly.embedding.SelfSupervisedEmbedding`
  - [x] Switch to pytorch lightning for training
  - [x] Add code to generate embeddings
- [x] Check that text is consistent with changes

### PR DESCRIPTION
**Pull request by guarin**
Closes #553

- Rewrite SimCLRModel
- Remove SelfSupervisedEmbedding
- Add lightning trainer
- Add embedding function

Outputs are the same as in previous version when training for 20 epochs.

[simclr_tutorial.pdf](https://github.com/lightly-ai/lightly/files/7568671/simclr_tutorial.pdf)

### CODE REVIEW

File: docs/source/tutorials_source/package/tutorial_simclr_clothing.py [hunk 1]
```diff
@@ -197,7 +233,8 @@ def get_image_as_np_array(filename: str):
     img = Image.open(filename)
     return np.asarray(img)
 
-def plot_knn_examples(embeddings, n_neighbors=3, num_examples=6):
+
+def plot_knn_examples(embeddings, filenames, n_neighbors=3, num_examples=6):
```
Comment (MalteEbner) [inline docs/source/tutorials_source/package/tutorial_simclr_clothing.py:237]: Nice that you made filenames an argument and don't use it from the outside.

File: docs/source/tutorials_source/package/tutorial_simclr_clothing.py [hunk 2]
```diff
@@ -290,18 +314,15 @@ def plot_knn_examples(embeddings, n_neighbors=3, num_examples=6):
...
 # load the model in a new file for inference
 resnet18_new = torchvision.models.resnet18()
-last_conv_channels = list(resnet.children())[-1].in_features
+
 # note that we need to create exactly the same backbone in order to load the weights
-backbone_new = nn.Sequential(
-    *list(resnet.children())[:-1],
-    nn.Conv2d(last_conv_channels, num_ftrs, 1),
-)
+backbone_new = nn.Sequential(*list(resnet18_new.children())[:-1])
```
Comment (MalteEbner) [inline docs/source/tutorials_source/package/tutorial_simclr_clothing.py:325]: Why is the `nn.Conv2d(last_conv_channels, num_ftrs, 1)` not needed anymore?

**Review by MalteEbner — APPROVED** [review]
This looks good to me.

File: docs/source/tutorials_source/package/tutorial_simclr_clothing.py (hunk 2 above)
Comment (guarin) [reply docs/source/tutorials_source/package/tutorial_simclr_clothing.py:325]: It is not part of the original SimCLR architecture and the network works without it.

**Pull request merged**